tools/test.py

```diff
@@ -299,6 +302,7 @@ def HasRun(self, output):
 
       if output.HasTimedOut():
         self.severity = 'fail'
+        self.traceback += '\nTimed Out'
```

Comment:
Not a big deal but I'd prefer we leave existing strings alone on this output where possible, it's parsable and we don't know what might depend on it--likely nothing but why risk it. I assume that in the majority of cases there's not going to be a traceback to append to so perhaps if you check for it being length>0 and then append `\n` and then regardless append 'timeout', so in the majority of cases we just get a 'timeout' and nothing's changed.

Reply:
Ping @Trott @refack

Reply:
P.S. `duration_ms` is consumed in two code paths, one treats it as seconds (the red arrow):
https://github.com/jenkinsci/tap-plugin/blob/05bd8843f0c87a379fdaec4f6e007c1124f1a35c/src/main/java/org/tap4j/plugin/model/TapStreamResult.java#L165-L167

and another place as milliseconds (the green arrow):
https://github.com/jenkinsci/tap-plugin/blob/3a80a472c6acff847ae3f572bb6bcc47d64ab9dd/src/main/java/org/tap4j/plugin/model/TapTestResultResult.java#L260-L271

![image](https://user-images.githubusercontent.com/96947/38470396-1f9d4706-3b30-11e8-9a91-337f74e493c1.png)

Reply:
P.P.S. I'm looking into finishing the work started by @jbergstroem (`tap2junit`) as the final goal.

Reply:
Rest of the changes look okay, but would it be okay to use `self.severity` to signal that the test has timed out? I think its better not to alter the traceback. Extreme corner case: if a test prints Timed Out and exits, we cannot be sure if the test exceeded the timelimit or not.

Reply:
Just for context the old code did
`severity` is the only really semantic field, so I would advise against chaging it.

Reply:
Just for context the old code did:
https://github.com/nodejs/node/blob/bdcd041890bf6b69b99a6b5870e16a5e85ecafd3/tools/test.py#L332

And even the current code does:
https://github.com/nodejs/node/blob/800ce94e5cdaa8dba7cf69b4c560a66a51ae0ec8/tools/test.py#L287
and
https://github.com/nodejs/node/blob/800ce94e5cdaa8dba7cf69b4c560a66a51ae0ec8/tools/test.py#L298
So we are already being naughty with `traceback`

---

As for `severity`, it is the only really semantic field (the TAP plugin uses it to categorize the results), so I would advise against changing it. A different option is to add a `message` field or even a dedicated `timed out` field.